mobile_verifier/src/reward_shares.rs

```diff
@@ -134,6 +141,65 @@ impl TransferRewards {
     }
 }
 
+#[derive(Default)]
+pub struct MapperShares {
+    pub discovery_mapping_shares: Vec<Vec<u8>>,
+    pub verification_mapping_shares: Vec<Vec<u8>>,
+}
+
+impl MapperShares {
+    pub fn new(active_subscribers: SubscriberMap, location_shares: LocationSharingMap) -> Self {
+        let discovery_mapping_shares = active_subscribers
+            .into_iter()
+            .filter(|(subscriber_id, data_transferred)| {
+                subscriber_shares_rewardable(&location_shares, subscriber_id, *data_transferred)
+            })
+            .map(|(subscriber_id, _)| subscriber_id)
+            .collect();
+        // default verificaiton mapping shares to empty vec, implemention tba
+        let verification_mapping_shares = Vec::new();
+        Self {
+            discovery_mapping_shares,
+            verification_mapping_shares,
+        }
+    }
+
+    pub fn rewards_per_share(
+        &self,
+        reward_period: &'_ Range<DateTime<Utc>>,
+    ) -> anyhow::Result<Decimal> {
+        let duration: Duration = reward_period.end - reward_period.start;
+        let total_mappers_pool = get_scheduled_tokens_for_mappers(duration);
+        let discovery_mappers_count = Decimal::from(self.discovery_mapping_shares.len());
+        let verification_mappers_count = Decimal::from(self.verification_mapping_shares.len());
+        let total_mapper_points = (discovery_mappers_count * DISCOVERY_MAPPING_POINTS_PER_SHARE)
+            + (verification_mappers_count * LOCATION_MAPPING_POINTS_PER_SHARE);
+        Ok(total_mappers_pool / total_mapper_points)
+    }
+
+    pub fn into_subscriber_rewards(
+        self,
+        reward_period: &'_ Range<DateTime<Utc>>,
+        reward_per_share: Decimal,
+    ) -> impl Iterator<Item = proto::MobileRewardShare> + '_ {
+        self.discovery_mapping_shares
+            .into_iter()
+            .map(move |subscriber_id| proto::SubscriberReward {
+                subscriber_id,
+                discovery_location_amount: (reward_per_share * DISCOVERY_MAPPING_POINTS_PER_SHARE)
```

Comment:
perhaps this is just a naming issue but multiplying these two variables seems to imply there's a missing component to the reward calculation to get the mobile bones in the `SubscriberReward`. i would expect to multiply two `_per_share` variables by a third `num_shares` or else `num_points * points_per_share` to get the number of shares and then multiply that by `reward_per_share` to get the reward bones